extension/src/basic-code-intel.ts

```diff
@@ -0,0 +1,47 @@
+import { Handler } from '@sourcegraph/basic-code-intel'
+import * as path from 'path'
+import * as sourcegraph from 'sourcegraph'
+
+export interface Providers {
+    hover: (doc: sourcegraph.TextDocument, pos: sourcegraph.Position) => Promise<sourcegraph.Hover | null>
+    definition: (doc: sourcegraph.TextDocument, pos: sourcegraph.Position) => Promise<sourcegraph.Definition | null>
+    references: (doc: sourcegraph.TextDocument, pos: sourcegraph.Position) => Promise<sourcegraph.Location[] | null>
+}
+
+export function initBasicCodeIntel(): Providers {
+    const handler = new Handler({
+        sourcegraph,
+        languageID: 'typescript',
+        fileExts: ['ts', 'tsx', 'js', 'jsx'],
+        commentStyle: {
+            lineRegex: /\/\/\s?/,
+            block: {
+                startRegex: /\/\*\*?/,
+                lineNoiseRegex: /(^\s*\*\s?)?/,
+                endRegex: /\*\//,
+            },
+        },
+        filterDefinitions: ({ filePath, fileContent, results }) => {
+            const imports = fileContent
+                .split('\n')
+                .map(line => {
+                    // Matches the import at index 1
+                    const match = /\bfrom ['"](.*)['"];?$/.exec(line) || /\brequire\(['"](.*)['"]\)/.exec(line)
+                    return match ? match[1] : undefined
+                })
+                .filter((x): x is string => Boolean(x))
+
+            const filteredResults = results.filter(result =>
+                imports.some(i => path.join(path.dirname(filePath), i) === result.file.replace(/\.[^/.]+$/, ''))
+            )
+
+            return filteredResults.length === 0 ? results : filteredResults
+        },
+    })
+
+    return {
+        hover: handler.hover.bind(handler),
+        definition: handler.definition.bind(handler),
+        references: handler.references.bind(handler),
+    }
```

Comment:
Can't this just return `handler`? Seems like that would be equivalent from a type perspective, and you could omit the `Providers` interface defined here and just use `Handlers` as return type (which is the real return type then)

(okay to keep if there is a reason)

Reply:
Oh yup, changed it so it returns `new Handler(...)` directly. The `.bind()` calls were for convenience in the Go extension, but are noops in the TypeScript extension, so I removed that. The `Providers` interface helps document what the intended use of this function is, and I removed the definition in this file and instead imported it from basic-code-intel.